lib/bibliothecary/analyser.rb

```diff
@@ -130,25 +130,43 @@ def analyse_file_info(file_info_list)
           .select(&method(:match_info?))
 
         matching_info.map do |info|
-          # TODO: Return multiple analysis for one info
-          analyse_contents_from_info(info)
-            .merge(related_paths: related_paths(info, matching_info))
+          analyse_contents_from_info(info).map do | analysis |
+            analysis.merge(related_paths: related_paths(info, matching_info))
+          end
         end
       end
 
       def analyse_contents(filename, contents)
         analyse_contents_from_info(FileInfo.new(nil, filename, contents))
       end
 
+
       def analyse_contents_from_info(info)
+        # If this Package Manager is a multiple, analyse multiple, otherwise just analyse one,
         kind = determine_kind_from_info(info)
+        kind.kind_of?(Array) ? analyse_multiple(kind, info) : analyse_one(kind, info)
+      end
+
+      def analyse_one(kind, info)
         dependencies = parse_file(info.relative_path, info.contents)
 
-        Bibliothecary::Analyser::create_analysis(platform_name, info.relative_path, kind, dependencies || [])
+        [create_analysis_for_dependencies(dependencies, info, kind)]
```

Comment:
I think we should really consider a way to do this without changing the API for every package manager.

Reply:
The previous way was making it "either a list or a hash of lists", but we had a meeting and discussed that it's better to make it be the same across the board, which unfortunately means we have to modify everything. We can have another meeting about this though if you'd like.